concent_api/core/payments/backends/sci_backend.py

```diff
@@ -55,6 +57,46 @@ def get_list_of_payments(
     return payments_list
 
 
+def are_all_payment_ts_younger_than_last_payment_closure_time_if_payment_exists(
+    requestor_eth_address: str,
+    provider_eth_address: str,
+    subtask_results_accepted_list: List[ForcedPaymentEvent],
+) -> bool:
+
+    oldest_payments_ts = _get_oldest_payment_timestamp_from_subtask_results_accepted_list(subtask_results_accepted_list)
+
+    forced_payment_event_list = _get_list_of_forced_payment_events(
+        requestor_eth_address=requestor_eth_address,
+        provider_eth_address=provider_eth_address,
+        search_payments_since_ts=oldest_payments_ts,
+    )
+
+    for forced_payment_event in forced_payment_event_list:
```

Comment:
This can be changed to 
```
return all([oldest_payments_ts < forced_payment_event.closure_time for forced_payment_event in forced_payment_event_list])
```

Reply:
but we are not interested in all `force_payment_events`, but it the last one ;)

Reply:
@dybi I agree.
It would be wrong because the comprehension returns `False` only if all the settlement payments found have `closure_time` after the oldest acceptance. Concent must reject the settlement request if even one of them satisfies this condition.

But I also agree with @rwrzesien that it should be checked with a compact list comprehension:
``` python
return not any([forced_payment_event.closure_time >= oldest_payments_ts for forced_payment_event in forced_payment_event_list])
```

Also, note that it should use `>=`, not `>`. The settlement payment settles acceptances up to `closure_time`, including that time.

Reply:
Implementation is changed, now function 'validates', so nothing is returned. It might only raise `BanksterTimestampError`

Reply:
Why it is raising exception from bankster? It should have it own, if any. I think it should return universal `True/False` than depend on exception.